judgments/views/advanced_search.py

```diff
@@ -1,179 +1,188 @@
-        context[key] = query_params[key] or ""
-
-    if errors.has_errors():
-        return TemplateResponse(
-            request, "pages/structured_search.html", context={"context": context}
-        )
-    else:
-        try:
-            query_without_stop_words = preprocess_query(query_text)
-            search_parameters = SearchParameters(
-                query=query_without_stop_words,
-                court=",".join(query_params["court"]),
-                judge=query_params["judge"],
-                party=query_params["party"],
-                page=int(page),
-                order=order,
-                date_from=query_params["from"],
-                date_to=query_params["to"],
-                page_size=int(per_page),
+@csrf_exempt
+def advanced_search(request):
+    """
+    The advanced search view handles any searches made in the application
+
+    * Given a valid search form, query Marklogic and return the results
+    * Given an invalid search form, render it again with the errors
+    * Given GET request without form submission return an empty form
+    * Given anything except an HTTP GET request raise an error
+    """
+    # We should only be handling GET requests here since we aren't changing anything on the server
+    if request.method == "GET":
+        form: AdvancedSearchForm = AdvancedSearchForm(request.GET)
+        params: dict = request.GET
+        """
+        Form should be valid unless there is a critical issue
+        with the submission (i.e. Month > 12)
+        """
+        if form.is_valid():
+            context: dict = {}
+            court_facets: dict = {}
+            year_facets: dict = {}
+            query_params: dict = {}
+            query_text: str = form.cleaned_data.get("query", "")
+            per_page: str = str(
+                as_integer(
+                    params.get("per_page", "10"),
+                    minimum=1,
+                    maximum=MAX_RESULTS_PER_PAGE,
+                    default=RESULTS_PER_PAGE,
+                )
             )
-            search_response = search_judgments_and_parse_response(
-                api_client, search_parameters
+            order: str = params.get("order", "-date")
+
+            from_date: Optional[date] = form.cleaned_data.get("from_date")
+            to_date: Optional[date] = form.cleaned_data.get("to_date", None)
+            # If a from_date is not specified, set it to the current min year
+            if not from_date:
+                from_date = date(CourtDates.min_year(), 1, 1)
+            else:
+                # Only provide the param back to the user if they set it
+                query_params = query_params | {
+                    "from": from_date,
+                    "from_date_0": from_date.day,
+                    "from_date_1": from_date.month,
+                    "from_date_2": from_date.year,
+                }
+            # If a to_date is not specified, set it to the current min year
```

Comment:
This comment is incorrect -- should be max year

Reply:
Concerned about searches missing records if the CourtDates are inaccurate -- can we not search with no dates instead?